src/system/unwind.c

```diff
@@ -132,34 +146,76 @@ void p2vrs_from_data_abort(_uw* sp, struct phase2_vrs* vrs) {
 	vrs->core.r[15] = sp[5] - 8;
 }
 
-// called by DataAbortInterrupt in rtos_hooks.c
-void report_data_abort(uint32_t _sp) {
+// called by DataAbortInterrupt and PrefetchAbortInterrupt in rtos_hooks.c
+//void report_data_abort(uint32_t _sp) {
+void report_fatal_error(uint32_t _sp, const char* error_name) {
 	struct phase2_vrs vrs;
 	p2vrs_from_data_abort((_uw*)_sp, &vrs);
 
-	fputs("\n\nDATA ABORT EXCEPTION\n\n", stderr);
+	fputs("\n\n", stderr);
+	fputs(error_name, stderr);
+	fputs("\n\n", stderr);
 	vexDisplayForegroundColor(ClrWhite);
 	vexDisplayBackgroundColor(ClrRed);
-	vexDisplayRectClear(0, 25, 480, 125);
-	vexDisplayString(2, "DATA ABORT EXCEPTION");
+	vexDisplayRectClear(0, 25, 480, 200);
+	vexDisplayString(2, error_name);
 	vexDisplayString(3, "PC: %x", vrs.core.r[R_PC]);
+
+	int brain_line_no = 4;
+	
 	if (pxCurrentTCB) {
-		vexDisplayString(4, "CURRENT TASK: %.32s\n", pxCurrentTCB->pcTaskName);
+		vexDisplayString(brain_line_no++, "CURRENT TASK: %.32s\n", pxCurrentTCB->pcTaskName);
 		fprintf(stderr, "CURRENT TASK: %.32s\n", pxCurrentTCB->pcTaskName);
 	}
 
-	fputs("REGISTERS AT ABORT\n", stderr);
+	fputs("REGISTERS AT ERROR\n", stderr);
 	print_phase2_vrs(&vrs);
 
 	fputs("BEGIN STACK TRACE\n", stderr);
 	fprintf(stderr, "\t%p\n", (void*)vrs.core.r[R_PC]);
-	__gnu_Unwind_Backtrace(trace_fn, NULL, &vrs);
+	struct trace_t trace = {{0}, 0, false};
+	__gnu_Unwind_Backtrace(trace_fn, &trace, &vrs);
 	fputs("END OF TRACE\n", stderr);
+	/*
```

Comment:
Everything works now, just get rid of this commented out code

Reply:
Done in https://github.com/purduesigbots/pros/pull/651/commits/c7d518be077a9387974f9a99ab316baf8d1086cd.